packages/ethash/src/index.ts

```diff
@@ -11,9 +11,127 @@ import {
 } from './util'
 // eslint-disable-next-line implicit-dependencies/no-implicit
 import type { LevelUp } from 'levelup'
-import type { Block, BlockHeader } from '@ethereumjs/block'
+import { Block, BlockData, BlockHeader, HeaderData } from '@ethereumjs/block'
 const xor = require('buffer-xor')
 
+type Solution = {
+  mixHash: Buffer
+  nonce: Buffer
+}
+
+class Miner {
+  private blockHeader: BlockHeader
+  private block?: Block
+  private ethash: Ethash
+
+  private currentNonce: BN
+  private solution?: Solution
```

Comment:
@jochem-brouwer I wouldn't want to unnecessarily expose all the class members and therefore make it part of the public API and rather keep all the above as `private`, have updated.

Let me know if there are good reasons to keep any of these as `public`.

Reply:
I'd say that the `solution` would be handy to keep `public`, then if a solution is found it can be taken from the Miner object directly. (One can also just re-call the miner function and just set the iterations to 0, it will immediately return the cache).

Reply:
Ok, I am preparing a release PR for the various libraries right now, will add a small fix along, guess that should be ok.